3 - FightClub/Code/FightClub.cs

```diff
@@ -7,17 +8,88 @@ public static class FightClubGame
 	{
 		public static IPowerUp PurchasePowerUp()
 		{
-			throw new NotImplementedException();
-		}
+            var powerUp = new PowerUp
+            {
+                DamageBuff = 1
+            };
+            return powerUp;
+        }
+
+
 
-		public static IPotion PurchasePotion()
+
+        public static IPotion PurchasePotion()
 		{
-			throw new NotImplementedException();
-		}
+            var potion = new Potion
+            {
+                Healing = 2,
+                Type = PotionType.Basic
+            };
+            return potion;
+        }
 
 		public static IFighter HireFighter()
 		{
-			throw new NotImplementedException();
+            var fighter = new Fighter
+            {
+                Damage = 1,
+                Health = 10
+            };
+            return fighter;
 		}
-	}
+
+        class PowerUp : IPowerUp
+        {
+            public int DamageBuff { get; set; }
+        }
+
+        class Potion : IPotion
+        {
+            public int Healing { get; set; }
+            public PotionType Type { get; set; }
+
+            public void SetType(PotionType potionType) 
+            {
+                if (Enum.Equals(potionType, PotionType.Greater))
+                { 
+                    Healing = 5; 
+                    Type = PotionType.Greater; 
+                }
+            }
+        }
+
+        class Fighter : IFighter
+        {
+            public int Health { get; set; }
+            public int Damage { get; set; }
+            public FighterState State { get; set; }
+
+            public void TakeDamage(IFighter fighter) 
+            {
+                if (Health >0)
+                { 
+                    Health = Health - fighter.Damage;
+                    if (Health < 10 && Health >1) State = FighterState.Hurt;
+                    if (Health == 1) State = FighterState.KnockedOut;
+                    if (Health == 0) State = FighterState.Dead;
```

Comment:
Be careful when trying to condense the code like you have, it can be hard to read for anyone coming into the code for the first time when its all squished. 

Secondly, I would advise doing else statements instead of three if statements. Not only are you processing over and over again, it is very easy for a bug to be introduced into here because a state can be overriden if someone maintains the code in the future.